### PR TITLE
CIAS30-3880

### PR DESCRIPTION
**Pull request by mimbor**
## Related tasks
- [CIAS30-3880](https://htdevelopers.atlassian.net/browse/CIAS30-3880)

## What's new?
- clear the selection of question to be reflected when the question with reflection is copied in the first place, before copying the reflection question 
- Reflections work after change of variable

### CODE REVIEW

**Review by WiktoriaRozanska — CHANGES_REQUESTED** [review]
Great job, I see a lot of new logic but without any new line in tests 🙈 

Additionally, even if I see in the logs that UpdateJobs::Adjust Question Reflections is done after fetching the question I still see the old variable, could you verify this?

[video with STR ](https://drive.google.com/file/d/1X6JnZKl7uGQ6PwNGDznKp3IbgkHxDrgi/view?usp=sharing)

**Comment by mimbor** [comment]
> Great job, I see a lot of new logic but without any new line in tests 🙈
> 
> Additionally, even if I see in the logs that UpdateJobs::Adjust Question Reflections is done after fetching the question I still see the old variable, could you verify this?
> 
> [video with STR ](https://drive.google.com/file/d/1X6JnZKl7uGQ6PwNGDznKp3IbgkHxDrgi/view?usp=sharing)

I have double checked it and after updating variable name, variable name in referenced questions is changed too

File: app/jobs/update_jobs/adjust_question_reflections.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      q.narrator['blocks'].each do |block|
+        next if block['reflections'].nil?
+
+        block['reflections'].each do |reflection|
+          next unless reflection['variable'] == prev_variable_name
+
+          reflection['variable'] = question.body['variable']['name']
+        end
+
+        q.save!
```
Comment (WiktoriaRozanska) [inline app/jobs/update_jobs/adjust_question_reflections.rb:15]: Here you can also use Postgresql operators/functions on jsonb to speed up a little bit this operation. It is a potential bottleneck

File: app/jobs/update_jobs/adjust_question_reflections.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      end
+    end
+  end
+
+  def reflectable_questions(question)
+    Question
+       .within_intervention(question.session.intervention_id)
+       .where.not(id: question.id)
+       .joins("CROSS JOIN jsonb_array_elements(narrator->'blocks') AS narrator_block")
```
Comment (WiktoriaRozanska) [inline app/jobs/update_jobs/adjust_question_reflections.rb:24]: I'm wondering if `CROSS JOIN` is necessary here.

File: spec/jobs/duplicate_jobs/session_spec.rb [hunk 3]
```diff
@@ -99,4 +99,82 @@
       expect(cloned_questions).to eq(['Question::Single'])
     end
   end
+
+  context 'when duplicating session with reflection' do
```
Comment (WiktoriaRozanska) [inline spec/jobs/duplicate_jobs/session_spec.rb:103]: What do you think about adding tests to detect if the source is the same in session_clone speck?

File: app/jobs/update_jobs/adjust_question_reflections.rb (hunk 2 above)
Comment (mimbor) [reply app/jobs/update_jobs/adjust_question_reflections.rb:24]: Unfortunatelly CROSS JOIN is necessary, because with outher joins this doesn't work

**Review by WiktoriaRozanska — COMMENTED** [review]
In the DB I still doesn't have updated value

**Pull request merged**